### PR TITLE
Add docs for DOLT_SCHEMA_DIFF

### PR DESCRIPTION
**Pull request by PavelSafronov**
Add docs for DOLT_SCHEMA_DIFF

### CODE REVIEW

File: content/reference/sql/version-control/dolt-sql-functions.md
Comment (fulghum) [inline content/reference/sql/version-control/dolt-sql-functions.md]: (minor) Not strictly required, but might be fun/cool/nice to try out the new embedded DoltHub Workbench UI that @liuliu-dev just added. There are some examples in the system tables page. It's cool because users get the read the docs and actually run queries and see results while they're learning about the functions. Cool to follow up with that separately if you need to stay focused on other CLI migration work though.

File: content/reference/sql/version-control/dolt-sql-functions.md
Comment (fulghum) [inline content/reference/sql/version-control/dolt-sql-functions.md]: Not docs related, but just looking at the function output... I'm wondering if we should be including the alter statements here. The output is just a little awkward with the same, long create table statements repeated on each line. Might be a cleaner interface to just have the table schemas, without repeating them, then do separate calls to `dolt_patch` to get the statements to run if needed. 

Would that have worked for the CLI migration you need this for? I know it would mean additional calls to the sql-server for the alter statements.

**Review by fulghum — APPROVED** [review]
Docs look good. Just one question about the function implementation itself and one minor/optional suggestion about using the embedded DoltHub workbench UI that Liuliu's been working on.

File: content/reference/sql/version-control/dolt-sql-functions.md
Comment (PavelSafronov) [reply content/reference/sql/version-control/dolt-sql-functions.md]: Yeah, on reflection, `dolt_patch` could have given us the required information. We would have ended up with a few more calls, and a bit of a refactor, but it would have worked. 😐

File: content/reference/sql/version-control/dolt-sql-functions.md
Comment (PavelSafronov) [reply content/reference/sql/version-control/dolt-sql-functions.md]: Working with Liu Liu on getting some schema changes into one of our example dolthub DBs, then will embed a query here.

Comment (fulghum) [reply content/reference/sql/version-control/dolt-sql-functions.md]: sweet!

File: content/reference/sql/version-control/dolt-sql-functions.md
Comment (fulghum) [reply content/reference/sql/version-control/dolt-sql-functions.md]: This function is new enough that we could still alter it without impacting customers (although we'd still want to call it out as a minor version just in case). It'll be **much** harder to change it in the future and we'll be stuck with this interface, so if we're not fully happy with it now, I think it's worth a couple hours to shift towards using `dolt_patch`. That's my vote, but if @macneale4 or @zachmu have a strong aversion to that, I could probably be convinced to just move on.

Comment (zachmu) [reply content/reference/sql/version-control/dolt-sql-functions.md]: I'm for not duplicating the dolt_patch functionality and keeping this narrowly scoped.

File: content/reference/sql/version-control/dolt-sql-functions.md
Comment (PavelSafronov) [reply content/reference/sql/version-control/dolt-sql-functions.md]: PR updated with an embed link with a DoltHub SQL query.
It *seems* to be comparable to other embed links, but I'm not sure. Is there a way to build docs locally?

Comment (liuliu-dev) [reply content/reference/sql/version-control/dolt-sql-functions.md]: I think the only way to see it will be merging it to `gitbook-dev` and then checking it on dev: https://dolt.gitbook.io/dolt-dev/sql-reference/version-control/dolt-system-tables
It might take a few minutes to sync to dev after merging the PR.

File: content/reference/sql/version-control/dolt-sql-functions.md
Comment (PavelSafronov) [reply content/reference/sql/version-control/dolt-sql-functions.md]: PR opened to drop the `alter_statement` column: https://github.com/dolthub/dolt/pull/6270
After that PR goes in, this PR will be updated to match the new functionality.

Comment (PavelSafronov) [reply content/reference/sql/version-control/dolt-sql-functions.md]: PR updated to match latest functionality, but the latest version of dolt is not deployed to DoltHub, so the embedded examples won't match the documentations.
PR converted to Draft and holding off on merging until the embedded examples match the new documentation.

Comment (PavelSafronov) [reply content/reference/sql/version-control/dolt-sql-functions.md]: Latest version deployed to DoltHub and the embedded doc link is working as expected:
https://www.dolthub.com/repositories/dolthub/docs_examples/embed/main?active=Tables&q=SELECT+*%0AFROM+dolt_schema_diff%28%27schema_diff_v1%27%2C+%27schema_diff_v2%27%29%3B%0A

**Review by fulghum — APPROVED** [review]
Looks great! 🚢

**Pull request merged**
